### DISCUSSION TITLE
Implementing bayesian spam/ham training

### DISCUSSION THREAD

**Discussion by onny** [emersion/maddy#637] [Q&A]
Hey,
I would like to train Rspamd for spam/ham mails. Most implementations I saw used Sieve rules to send and copy mails to specific mailbox and IMAP directory. An additional script fetches the "reports" and feeds them into Rspamd [1].

Has someone managed to do this in Maddy? Is there an alternative for Sieve rules to process mails when moved into or out of spam folder?

[1] https://pipo.blog/articles/20210311-rspamd-trainer-bayes

**Comment by foxcpp** [comment]
No, currently there are no hooks implemented that would allow such integration.
